python/ray/train/gbdt_trainer.py

```diff
@@ -224,4 +224,6 @@ def default_resource_request(cls, config):
                     scaling_config_dataclass, ray_params_cls, default_ray_params
                 ).get_tune_resources()
 
+        GBDTTrainable.__name__ = trainer_cls.__name__
```

Comment:
this is a bit hacky; can we instead have it render in the __repr__ of the class only?

Reply:
This is what we are doing in DataParallelTrainer as well (and indeed, what happens inside `wrap_trainable` - and this class inherits from the return of that). I honestly think it's fine

Reply:
actually, sure, let me change that

Reply:
ok so this is odd but `__repr__` is already set correctly (as it inherits from `wrap_function`). Not sure why it's not showing up in logs as intended.